File: extensions/spark/kyuubi-spark-connector-tpcds/src/main/java/io/trino/tpcds/row/CustomerDemographicsRow.java
package io.trino.tpcds.row;

import static io.trino.tpcds.generator.CustomerDemographicsGeneratorColumn.CD_CREDIT_RATING;
import static io.trino.tpcds.generator.CustomerDemographicsGeneratorColumn.CD_DEMO_SK;
import static io.trino.tpcds.generator.CustomerDemographicsGeneratorColumn.CD_DEP_COLLEGE_COUNT;
import static io.trino.tpcds.generator.CustomerDemographicsGeneratorColumn.CD_DEP_COUNT;
import static io.trino.tpcds.generator.CustomerDemographicsGeneratorColumn.CD_DEP_EMPLOYED_COUNT;
import static io.trino.tpcds.generator.CustomerDemographicsGeneratorColumn.CD_EDUCATION_STATUS;
import static io.trino.tpcds.generator.CustomerDemographicsGeneratorColumn.CD_GENDER;
import static io.trino.tpcds.generator.CustomerDemographicsGeneratorColumn.CD_MARITAL_STATUS;
import static io.trino.tpcds.generator.CustomerDemographicsGeneratorColumn.CD_PURCHASE_ESTIMATE;

import org.apache.kyuubi.spark.connector.tpcds.row.KyuubiTPCDSTableRowWithNulls;

public class CustomerDemographicsRow extends KyuubiTPCDSTableRowWithNulls {
  private final long cdDemoSk;
  private final String cdGender;
  private final String cdMaritalStatus;
  private final String cdEducationStatus;
  private final int cdPurchaseEstimate;
  private final String cdCreditRating;
  private final int cdDepCount;
  private final int cdDepEmployedCount;
  private final int cdDepCollegeCount;

  public CustomerDemographicsRow(
      long nullBitMap,
      long cdDemoSk,
      String cdGender,
      String cdMaritalStatus,
      String cdEducationStatus,
      int cdPurchaseEstimate,
      String cdCreditRating,
      int cdDepCount,
      int cdDepEmployedCount,
      int cdDepCollegeCount) {
    super(nullBitMap, CD_DEMO_SK);
    this.cdDemoSk = cdDemoSk;
    this.cdGender = cdGender;
    this.cdMaritalStatus = cdMaritalStatus;
    this.cdEducationStatus = cdEducationStatus;
    this.cdPurchaseEstimate = cdPurchaseEstimate;
    this.cdCreditRating = cdCreditRating;
    this.cdDepCount = cdDepCount;
    this.cdDepEmployedCount = cdDepEmployedCount;
    this.cdDepCollegeCount = cdDepCollegeCount;
  }

  public long getCdDemoSk() {
    return cdDemoSk;
  }

  public String getCdGender() {
    return cdGender;
  }

  public String getCdMaritalStatus() {
    return cdMaritalStatus;
  }

  public String getCdEducationStatus() {
    return cdEducationStatus;
  }

  public int getCdPurchaseEstimate() {
    return cdPurchaseEstimate;
  }

  public String getCdCreditRating() {
    return cdCreditRating;
  }

  public int getCdDepCount() {
    return cdDepCount;
  }

  public int getCdDepEmployedCount() {
    return cdDepEmployedCount;
  }

  public int getCdDepCollegeCount() {
    return cdDepCollegeCount;
  }

  @Override
  public Object[] values() {
    return new Object[] {
      getOrNullForKey(cdDemoSk, CD_DEMO_SK),
      getOrNull(cdGender, CD_GENDER),
      getOrNull(cdMaritalStatus, CD_MARITAL_STATUS),
      getOrNull(cdEducationStatus, CD_EDUCATION_STATUS),
      getOrNull(cdPurchaseEstimate, CD_PURCHASE_ESTIMATE),
      getOrNull(cdCreditRating, CD_CREDIT_RATING),
      getOrNull(cdDepCount, CD_DEP_COUNT),
      getOrNull(cdDepEmployedCount, CD_DEP_EMPLOYED_COUNT),
      getOrNull(cdDepCollegeCount, CD_DEP_COLLEGE_COUNT)
    };
  }
}

File: extensions/spark/kyuubi-spark-connector-tpcds/src/main/scala/io/trino/tpcds/row/CustomerDemographicsRow.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.trino.tpcds.row

import io.trino.tpcds.generator.CustomerDemographicsGeneratorColumn._
import org.apache.spark.sql.catalyst.expressions.GenericInternalRow
import org.apache.spark.sql.catalyst.InternalRow

import java.util.{List => JList}

class CustomerDemographicsRow(
                               nullBitMap: Long,
                               private val cdDemoSk: Long,
                               private val cdGender: String,
                               private val cdMaritalStatus: String,
                               private val cdEducationStatus: String,
                               private val cdPurchaseEstimate: Int,
                               private val cdCreditRating: String,
                               private val cdDepCount: Int,
                               private val cdDepEmployedCount: Int,
                               private val cdDepCollegeCount: Int

                               ) extends TableRowWithNulls(nullBitMap, CD_DEMO_SK)
  with KyuubiTableRowWithNulls {

    override val nullBitMapInternal = nullBitMap
    override val firstColumnInternal = CD_DEMO_SK

  def getCdDemoSk: Long = cdDemoSk
  def getCdGender: String = cdGender
  def getCdMaritalStatus: String = cdMaritalStatus
  def getCdEducationStatus: String = cdEducationStatus
  def getCdPurchaseEstimate: Int = cdPurchaseEstimate
  def getCdCreditRating: String = cdCreditRating
  def getCdDepCount: Int = cdDepCount
  def getCdDepEmployedCount: Int = cdDepEmployedCount
  def getCdDepCollegeCount: Int = cdDepCollegeCount

  override def getValues: JList[String] = throw new UnsupportedOperationException()

  def internalRow: InternalRow =
    new GenericInternalRow(
      Array(
        getLongOrNull(cdDemoSk, CD_DEMO_SK),
        getStringOrNullInternal(cdGender, CD_GENDER),
        getStringOrNullInternal(cdMaritalStatus, CD_MARITAL_STATUS),
        getStringOrNullInternal(cdEducationStatus, CD_EDUCATION_STATUS),
        getIntOrNull(cdPurchaseEstimate, CD_PURCHASE_ESTIMATE),
        getStringOrNullInternal(cdCreditRating, CD_CREDIT_RATING),
        getIntOrNull(cdDepCount, CD_DEP_COUNT),
        getIntOrNull(cdDepEmployedCount, CD_DEP_EMPLOYED_COUNT),
        getIntOrNull(cdDepCollegeCount, CD_DEP_COLLEGE_COUNT)))
}
